commit 877818802c3e970f67ccb53012facc78bef5f97a
Author: Borislav Petkov (AMD) <bp@alien8.de>
Date:   Mon Nov 11 17:22:08 2024 +0100

    x86/bugs: Add SRSO_USER_KERNEL_NO support
    
    If the machine has:
    
      CPUID Fn8000_0021_EAX[30] (SRSO_USER_KERNEL_NO) -- If this bit is 1,
      it indicates the CPU is not subject to the SRSO vulnerability across
      user/kernel boundaries.
    
    have it fall back to IBPB on VMEXIT only, in the case it is going to run
    VMs:
    
      Speculative Return Stack Overflow: Mitigation: IBPB on VMEXIT only
    
    Signed-off-by: Borislav Petkov (AMD) <bp@alien8.de>
    Reviewed-by: Nikolay Borisov <nik.borisov@suse.com>
    Link: https://lore.kernel.org/r/20241202120416.6054-2-bp@kernel.org

diff --git a/arch/x86/kernel/cpu/common.c b/arch/x86/kernel/cpu/common.c
index 3e9037690814..7e8d811b51c8 100644
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -1270,6 +1270,7 @@ static const struct x86_cpu_id cpu_vuln_blacklist[] __initconst = {
 	VULNBL_AMD(0x17, RETBLEED | SMT_RSB | SRSO),
 	VULNBL_HYGON(0x18, RETBLEED | SMT_RSB | SRSO),
 	VULNBL_AMD(0x19, SRSO),
+	VULNBL_AMD(0x1a, SRSO),
 	{}
 };